app/admin/Settings.scala

```diff
@@ -17,19 +24,50 @@ case class Settings(
 )
 
 object Settings {
-  def fromConfig(config: Config): Settings =
-    Settings(
-      Switches(
-        PaymentMethodsSwitch.fromConfig(config.getConfig("oneOff")),
-        PaymentMethodsSwitch.fromConfig(config.getConfig("recurring")),
-        experimentsFromConfig(config, "abtests"),
-        SwitchState.fromConfig(config, "optimize"),
-        SwitchState.fromConfig(config, "internationalSubscribePages")
-      )
-    )
+  def fromDiskOrS3(config: Config, s3: AmazonS3): Either[Throwable, Settings] = {
+    val adminSettingsSource: Either[Throwable, AdminSettingsSource] = if (config.hasPath("local.path")) {
+      AdminSettingsSource.fromDisk(config.getString("local.path"))
```

Comment:
`getString()` can throw an exception; so the function signature of `fromDiskOrS3` is misrepresentative - we're not always encoding the error as a value